### PR TITLE
PP-10864 remove smartpay part 3

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- remove the rest of smartpay code

### CODE REVIEW

**Review by james-peacock-ho — DISMISSED** [review]
LGTM

**Review by nlsteers — DISMISSED** [review]
Pinning this while we fix `staging-2`

File: src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java [hunk 1]
```diff
@@ -18,6 +19,7 @@
 import javax.validation.constraints.NotNull;
 import java.time.Duration;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java]: Any reason we need to ignore unknown properties here?

File: src/test/java/uk/gov/pay/connector/it/gatewayclient/ClientFactoryIT.java [hunk 2]
```diff
@@ -183,26 +180,6 @@ public void shouldUseGatewaySpecificReadTimeoutOverride_whenSpecified() {
         assertGatewayFailure(request, authOverriddenTimeout);
     }
 
-    @Test
-    public void shouldUseGatewaySpecificReadTimeoutOverrideForSmartpay_whenSpecified() {
-        app = startApp("config/client-factory-test-config-with-smartpay-timeout-override.yaml", false);
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/gatewayclient/ClientFactoryIT.java:188]: Can file `client-factory-test-config-with-smartpay-timeout-override.yaml` be removed too ?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountSearchParams.java [hunk 3]
```diff
@@ -86,10 +86,10 @@ public class GatewayAccountSearchParams {
     private String type;
 
     @QueryParam("payment_provider")
-    @Pattern(regexp = "sandbox|worldpay|smartpay|epdq|stripe",
-            message = "Parameter [payment_provider] must be one of 'sandbox', 'worldpay', 'smartpay', 'epdq' or 'stripe'")
+    @Pattern(regexp = "sandbox|worldpay|epdq|stripe",
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountSearchParams.java]: We still have smartpay LIVE accounts that have not switched PSP. We need this longer to be able to search in Toolbox

File: src/test/resources/config/client-factory-test-config-with-smartpay-timeout-override.yaml
Comment (kbottla) [inline src/test/resources/config/client-factory-test-config-with-smartpay-timeout-override.yaml:1]: This file can be removed entirely as it is no longer referenced anywhere

**Review by kbottla — DISMISSED** [review]

File: src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java]: 👍

File: src/test/java/uk/gov/pay/connector/it/gatewayclient/ClientFactoryIT.java (hunk 2 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/gatewayclient/ClientFactoryIT.java:188]: 👍

File: src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountSearchParams.java (hunk 3 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountSearchParams.java]: 👍

File: src/test/resources/config/client-factory-test-config-with-smartpay-timeout-override.yaml
Comment (SandorArpa) [reply src/test/resources/config/client-factory-test-config-with-smartpay-timeout-override.yaml:1]: 👍

File: src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java [hunk 4]
```diff
@@ -4,14 +4,7 @@
...
 import io.dropwizard.db.DataSourceFactory;
-import uk.gov.pay.connector.app.config.Authorisation3dsConfig;
-import uk.gov.pay.connector.app.config.AuthorisationConfig;
-import uk.gov.pay.connector.app.config.EmittedEventSweepConfig;
-import uk.gov.pay.connector.app.config.EventEmitterConfig;
-import uk.gov.pay.connector.app.config.ExpungeConfig;
-import uk.gov.pay.connector.app.config.PayoutReconcileProcessConfig;
-import uk.gov.pay.connector.app.config.RestClientConfig;
-import uk.gov.pay.connector.app.config.TaskQueueConfig;
+import uk.gov.pay.connector.app.config.*;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java]: Wildcard import?

Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java]: 👍

**Review by kbottla — APPROVED** [review]

**Pull request merged**